soc/arm/nxp_imx/mcimx8mm_a53/mmu_regions.c

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2020 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <soc.h>
+#include <arch/arm/aarch64/arm_mmu.h>
+
+#define SZ_1K	1024
+
+static const struct arm_mmu_region mmu_regions[] = {
+
+	MMU_REGION_FLAT_ENTRY("GIC",
+			      DT_REG_ADDR_BY_IDX(DT_INST(0, arm_gic), 0),
+			      DT_REG_SIZE_BY_IDX(DT_INST(0, arm_gic), 0),
+			      MT_DEVICE_nGnRnE | MT_RW | MT_NS),
+
+	MMU_REGION_FLAT_ENTRY("GIC",
+			      DT_REG_ADDR_BY_IDX(DT_INST(0, arm_gic), 1),
+			      DT_REG_SIZE_BY_IDX(DT_INST(0, arm_gic), 1),
+			      MT_DEVICE_nGnRnE | MT_RW | MT_NS),
+
+	MMU_REGION_FLAT_ENTRY("UART",
+			      DT_REG_ADDR(DT_INST(0, nxp_imx_iuart)),
+			      DT_REG_SIZE(DT_INST(0, nxp_imx_iuart)),
+			      MT_DEVICE_nGnRnE | MT_RW | MT_NS),
+
+	MMU_REGION_FLAT_ENTRY("CCM",
+			      DT_REG_ADDR(DT_INST(0, nxp_imx_ccm)),
+			      DT_REG_SIZE(DT_INST(0, nxp_imx_ccm)),
+			      MT_DEVICE_nGnRnE | MT_RW | MT_NS),
+
+
+	MMU_REGION_FLAT_ENTRY("ANA_PLL",
+			      DT_REG_ADDR(DT_INST(0, nxp_imx_ana)),
+			      DT_REG_SIZE(DT_INST(0, nxp_imx_ana)),
+			      MT_DEVICE_nGnRnE | MT_RW | MT_NS),
+
+	MMU_REGION_FLAT_ENTRY("DRAM",
+			      CONFIG_SRAM_BASE_ADDRESS,
```

Comment:
DRAM vs SRAM. But why do you need this? In general it should not be needed to map the whole SRAM.

Reply:
I am thinking we need memory for IPC usage, between zephyr and Linux.

Reply:
Ok, but mapping the whole SRAM is probably an overkill. Can you define a memory region to be used for IPC and map that instead?

Reply:
sure. Currently I have not add that piece memory, will add that in future PRs.